### PR TITLE
Added `mimirtool rules delete-namespace`

### PR DESCRIPTION
**Pull request by wilfriedroset**
#### What this PR does
The following changes allow user delete all the rule groups in a namespace (including the namespace itself).

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pracucci** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

**Comment by wilfriedroset** [comment]
Nice this means mimir 2.4 is coming 👀 
PR has been rebase and CHANGELOG.md updated accordingly.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 ### Jsonnet
 
 ### Mimirtool
+* [ENHANCEMENT] Added `mimirtool rules delete-namespace` command delete all the rule groups in a namespace including the namespace itself. #3136
```
Comment (osg-grafana) [inline CHANGELOG.md:14]:
```suggestion
* [ENHANCEMENT] Added `mimirtool rules delete-namespace` command to delete all of the rule groups in a namespace including the namespace itself. #3136
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 2]
```diff
@@ -196,6 +196,14 @@ groups:
         expr: sum by (job) (http_inprogress_requests)
 ```
 
+#### Delete a namespace
+
+The following command deletes all the rule groups in a namespace (including the namespace itself).
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:201]:
```suggestion
The following command deletes all of the rule groups in a namespace, including the namespace itself.
```

**Review by osg-grafana — APPROVED** [review]
Unblocking with nits

**Review by 56quarters — APPROVED** [review]
Code looks good to me. I'll merge once @osg-grafana's feedback is addressed. Thanks!

**Pull request merged**